### ISSUE TITLE
Delve freezes/stops when stepping into breakpoint

### ISSUE THREAD

**Issue by chancez** [derekparker/delve#342]
https://gist.github.com/ecnahc515/74d85ca61e1decb003ab

This is from the `wip` branch on my fork of the `authclub/billforward` bindings: https://github.com/ecnahc515/billforward/tree/wip

I ran `delve debug --log=true` within the `example/` directory of the repo.

Version info:
```
Delve Debugger
Version: 0.10.0-alpha
Build: d4bfd25a28cab4589808471d987e28ae6c6175c6
```
Go 1.5:
```
go version go1.5.2 darwin/amd64
```

OS X Yosemite 10.10.5 (14F27)

**Comment by ecnahc515** [comment]
Here's the stack trace after running `pkill -ABRT dlv`:

```
goroutine 0 [idle]:
runtime.mach_semaphore_wait(0xd03, 0x3603, 0xc800000000, 0x40182f2, 0x401456f, 0x4937900, 0x4055d89, 0xffffffffffffffff, 0xc820560480, 0x7fff5fbff3d4, ...)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/sys_darwin_amd64.s:407 +0xb
runtime.semasleep1(0xffffffffffffffff, 0xc820560480)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/os1_darwin.go:395 +0xe5
runtime.semasleep.func1()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/os1_darwin.go:411 +0x29
runtime.systemstack(0x7fff5fbff3d8)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/asm_amd64.s:278 +0xab
runtime.semasleep(0xffffffffffffffff, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/os1_darwin.go:412 +0x36
runtime.notesleep(0x49381d0)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/lock_sema.go:169 +0x100
runtime.stoplockedm()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/proc1.go:1276 +0xae
runtime.schedule()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/proc1.go:1598 +0x6e
runtime.park_m(0xc820001680)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/proc1.go:1706 +0x18b
runtime.mcall(0x7fff5fbff4d0)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/asm_amd64.s:204 +0x5b

goroutine 18 [chan receive, 5 minutes, locked to thread]:
runtime.gopark(0x4752308, 0xc82011e718, 0x465f370, 0xc, 0x17, 0x3)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/proc.go:185 +0x163 fp=0xc820030640 sp=0xc820030618
runtime.goparkunlock(0xc82011e718, 0x465f370, 0xc, 0x45bb817, 0x3)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/proc.go:191 +0x54 fp=0xc820030678 sp=0xc820030640
runtime.chanrecv(0x4492e60, 0xc82011e6c0, 0xc820030788, 0x41cf001, 0xc820a70000)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/chan.go:448 +0x472 fp=0xc820030738 sp=0xc820030678
runtime.chanrecv2(0x4492e60, 0xc82011e6c0, 0xc820030788, 0x101)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/chan.go:354 +0x2b fp=0xc820030768 sp=0xc820030738
github.com/derekparker/delve/proc.(*Process).handlePtraceFuncs(0xc820144000)
	/Users/chance/go/src/github.com/derekparker/delve/proc/proc.go:696 +0xc3 fp=0xc8200307b8 sp=0xc820030768
runtime.goexit()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/asm_amd64.s:1721 +0x1 fp=0xc8200307c0 sp=0xc8200307b8
created by github.com/derekparker/delve/proc.New
	/Users/chance/go/src/github.com/derekparker/delve/proc/proc.go:71 +0x1d4

goroutine 1 [chan receive, 5 minutes]:
net/rpc.(*Client).Call(0xc820384000, 0xc820c024c0, 0x11, 0x447ba00, 0xc820c024a0, 0x447ba60, 0xc820269d40, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/client.go:315 +0xcd
github.com/derekparker/delve/service/rpc.(*RPCClient).call(0xc820ba0060, 0x464a3a8, 0x7, 0x447ba00, 0xc820c024a0, 0x447ba60, 0xc820269d40, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/service/rpc/client.go:234 +0xb4
github.com/derekparker/delve/service/rpc.(*RPCClient).Step(0xc820ba0060, 0x47519b8, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/service/rpc/client.go:86 +0xcc
github.com/derekparker/delve/terminal.step(0xc8202ea6e0, 0x0, 0x0, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/terminal/command.go:469 +0x4c
github.com/derekparker/delve/terminal.(*Term).Run(0xc8202ea6e0, 0x0, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/terminal/terminal.go:110 +0xab6
main.execute(0x0, 0xc820110990, 0x1, 0x1, 0xc820112058, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/cmd/dlv/main.go:340 +0x598
main.main.func3.1(0xc8201107c0, 0x0, 0x1, 0xc820112058, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/cmd/dlv/main.go:103 +0x45c
main.main.func3(0xc82013c5a0, 0xc8201107c0, 0x0, 0x1)
	/Users/chance/go/src/github.com/derekparker/delve/cmd/dlv/main.go:87 +0x3e
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).execute(0xc82013c5a0, 0xc820110770, 0x1, 0x1, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:547 +0x869
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).Execute(0xc82013c000, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:630 +0x46a
main.main()
	/Users/chance/go/src/github.com/derekparker/delve/cmd/dlv/main.go:294 +0xd0f

goroutine 17 [syscall, 5 minutes, locked to thread]:
runtime.goexit()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [syscall, 5 minutes]:
os/signal.loop()
	/usr/local/Cellar/go/1.5.2/libexec/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/Cellar/go/1.5.2/libexec/src/os/signal/signal_unix.go:28 +0x37

goroutine 32 [IO wait, 5 minutes]:
net.runtime_pollWait(0x4ca12e0, 0x72, 0xc8200101e0)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820853720, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820853720, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8208536c0, 0xc820ce2601, 0x5ff, 0x5ff, 0x0, 0x4c60050, 0xc8200101e0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820cce000, 0xc820ce2601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/net.go:172 +0xe4
encoding/json.(*Decoder).refill(0xc8202fa000, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/encoding/json/stream.go:152 +0x287
encoding/json.(*Decoder).readValue(0xc8202fa000, 0x1, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/encoding/json/stream.go:128 +0x41b
encoding/json.(*Decoder).Decode(0xc8202fa000, 0x4569320, 0xc8203c6048, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/encoding/json/stream.go:57 +0x159
net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0xc8203c6000, 0xc820362180, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/jsonrpc/client.go:75 +0x8f
net/rpc.(*Client).input(0xc820384000)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/client.go:109 +0xbf
created by net/rpc.NewClientWithCodec
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/client.go:201 +0xd2

goroutine 7 [IO wait, 5 minutes]:
net.runtime_pollWait(0x4ca1220, 0x72, 0xc8200101e0)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82029c060, 0x72, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82029c060, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82029c000, 0xc820ce2001, 0x5ff, 0x5ff, 0x0, 0x4c60050, 0xc8200101e0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820032000, 0xc820ce2001, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/net.go:172 +0xe4
encoding/json.(*Decoder).refill(0xc820068340, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/encoding/json/stream.go:152 +0x287
encoding/json.(*Decoder).readValue(0xc820068340, 0x1, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/encoding/json/stream.go:128 +0x41b
encoding/json.(*Decoder).Decode(0xc820068340, 0x45693e0, 0xc820156020, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/encoding/json/stream.go:57 +0x159
net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0xc820156000, 0xc82076e060, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/jsonrpc/server.go:66 +0x7c
net/rpc.(*Server).readRequestHeader(0xc820269740, 0x62cc8e0, 0xc820156000, 0x0, 0x0, 0xc82076e060, 0xc820353c00, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/server.go:576 +0x90
net/rpc.(*Server).readRequest(0xc820269740, 0x62cc8e0, 0xc820156000, 0xc820269740, 0xc820a740a0, 0xc82024c100, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/server.go:543 +0x8b
net/rpc.(*Server).ServeCodec(0xc820269740, 0x62cc8e0, 0xc820156000)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/server.go:462 +0x8c
github.com/derekparker/delve/service/rpc.(*RPCServer).Run.func1(0xc820114b80)
	/Users/chance/go/src/github.com/derekparker/delve/service/rpc/server.go:65 +0x485
created by github.com/derekparker/delve/service/rpc.(*RPCServer).Run
	/Users/chance/go/src/github.com/derekparker/delve/service/rpc/server.go:66 +0x118

goroutine 33 [select, 5 minutes, locked to thread]:
runtime.gopark(0x4752340, 0xc82054a728, 0x46519e0, 0x6, 0x4033018, 0x2)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc82054a728, 0x0, 0x18)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc82054a728)
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/signal1_unix.go:227 +0x323
runtime.goexit()
	/usr/local/Cellar/go/1.5.2/libexec/src/runtime/asm_amd64.s:1721 +0x1

goroutine 50 [chan receive, 5 minutes]:
github.com/derekparker/delve/terminal.(*Term).Run.func1(0xc820384240, 0xc8202ea6e0)
	/Users/chance/go/src/github.com/derekparker/delve/terminal/terminal.go:49 +0x6b
created by github.com/derekparker/delve/terminal.(*Term).Run
	/Users/chance/go/src/github.com/derekparker/delve/terminal/terminal.go:55 +0x1c6

goroutine 67 [syscall, 5 minutes, locked to thread]:
github.com/derekparker/delve/proc._Cfunc_mach_port_wait(0xc800002403, 0x0)
	??:0 +0x35
github.com/derekparker/delve/proc.(*Process).trapWait(0xc820144000, 0x0, 0x7fff86eef232, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/proc/proc_darwin.go:271 +0x4b
github.com/derekparker/delve/proc.(*Thread).singleStep(0xc820ae6fc0, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/proc/threads_darwin.go:30 +0x9d
github.com/derekparker/delve/proc.(*Thread).Step(0xc820ae6fc0, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/proc/threads.go:92 +0x1a7
github.com/derekparker/delve/proc.(*Process).Step.func1(0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/proc/proc.go:406 +0xe1
github.com/derekparker/delve/proc.(*Process).run(0xc820144000, 0xc820045888, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/proc/proc.go:684 +0x123
github.com/derekparker/delve/proc.(*Process).Step(0xc820144000, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/proc/proc.go:413 +0x53
github.com/derekparker/delve/service/debugger.(*Debugger).Command(0xc820110a10, 0xc820c02560, 0x18, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/service/debugger/debugger.go:255 +0x5a6
github.com/derekparker/delve/service/rpc.(*RPCServer).Command(0xc820114b80, 0xc820c02560, 0xc820269d80, 0x0, 0x0)
	/Users/chance/go/src/github.com/derekparker/delve/service/rpc/server.go:96 +0x3f
reflect.Value.call(0x4547ea0, 0x462d6c0, 0x13, 0x464a208, 0x4, 0xc820045ef8, 0x3, 0x3, 0x0, 0x0, ...)
	/usr/local/Cellar/go/1.5.2/libexec/src/reflect/value.go:432 +0x120a
reflect.Value.Call(0x4547ea0, 0x462d6c0, 0x13, 0xc820045ef8, 0x3, 0x3, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.2/libexec/src/reflect/value.go:300 +0xb1
net/rpc.(*service).call(0xc820269780, 0xc820269740, 0xc820a740a0, 0xc82024c100, 0xc820ba0820, 0x447ba00, 0xc820c02560, 0x16, 0x447ba60, 0xc820269d80, ...)
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/server.go:383 +0x1c1
created by net/rpc.(*Server).ServeCodec
	/usr/local/Cellar/go/1.5.2/libexec/src/net/rpc/server.go:477 +0x4ac

rax    0xe
rbx    0xd03
rcx    0x7fff5fbff360
rdx    0x7fff5fbff3d8
rdi    0xd03
rsi    0x4937900
rbp    0x49380c0
rsp    0x7fff5fbff360
r8     0x49380c0
r9     0x0
r10    0x0
r11    0x286
r12    0x8
r13    0x474e132
r14    0x9
r15    0x8
rip    0x40630bb
rflags 0x286
cs     0x7
fs     0x0
gs     0x0
```

**Comment by aarzilli** [comment]
I've tested this with `step` and `step-instruction` and I can't reproduce it, I think it was fixed by 39bc379dd29f72138c1f7587fe42925d68ca1469.
Reopen if you still encounter this issue. Thank you.
